### PR TITLE
Add check for same number of pattern and posix-pattern statements

### PR DESCRIPTION
**Pull request by wenovus**

### CODE REVIEW

File: openconfig_pyang/plugins/openconfig.py [hunk 1]
```diff
@@ -721,6 +729,29 @@ def check_enumeration_style(ctx, stmt):
...
+    Args:
+      ctx: pyang.Context for validation
+      stmt: pyang.Statement representing a leaf or leaf-list
+          containing an enumeration
+    """
+    elemtype = stmt.search_one("type")
+    if elemtype is None:
+      return
+
+    patternCount = len(elemtype.search("pattern"))
```
Comment (robshakir) [inline openconfig_pyang/plugins/openconfig.py:746]: Does `elemtype.search` return `None` if there is no pattern statement? If so, doesn't `len(None)` throw a `TypeError`?

File: tests/oclinter/unequal-posix-pattern/Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+ok:
+	pyang --plugindir $(PLUGIN_DIR) \
+		--openconfig --oc-only -p ${ROOT_DIR}/../common \
+		${ROOT_DIR}/openconfig-testcase-succeed.yang
+
+broken:
+	pyang --plugindir $(PLUGIN_DIR) \
+	    --openconfig --oc-only -p ${ROOT_DIR}/../common \
+			    ${ROOT_DIR}/openconfig-testcase-fail.yang
```
Comment (robshakir) [inline tests/oclinter/unequal-posix-pattern/Makefile]: (very nitty-nit) Can we align the indentation here?

File: tests/oclinter/unequal-posix-pattern/openconfig-testcase-fail.yang [hunk 3]
```diff
@@ -0,0 +1,41 @@
+module openconfig-testcase-fail {
+  prefix "oc-tc";
+  namespace "http://openconfig.net/linter/testcase";
+
+  import openconfig-extensions { prefix oc-ext; }
+
+  description
+    "Failure test case for using a bad type within
+    an OpenConfig mdoel.";
```
Comment (robshakir) [inline tests/oclinter/unequal-posix-pattern/openconfig-testcase-fail.yang]: `s/mdoel/model/`

**Review by robshakir — APPROVED** [review]
One case that I'm not sure about - which might be worth covering in your testcase modules, but presuming that all the other testcases pass it might be that I didn't quite understand what the `search` API returns.

File: openconfig_pyang/plugins/openconfig.py (hunk 1 above)
Comment (wenovus) [reply openconfig_pyang/plugins/openconfig.py:746]: Just checked that it always returns a list.

**Pull request merged**